src/Sentry.Unity.Editor/Native/SentryWindowsPlayer.cs

```diff
@@ -0,0 +1,140 @@
+
+            // TODO: Does the .props file have to look like that?
+            using var props = File.CreateText(Path.Combine(windowsPlayerTarget, "UnityCommon.props"));
+            props.Write(@"<?xml version=""1.0"" encoding=""utf-8""?>
+<Project xmlns=""http://schemas.microsoft.com/developer/msbuild/2003"">
+</Project>");
+        }
+
+        internal static string LocateMSBuild(string vsWherePath, IDiagnosticLogger? logger)
+        {
+            StartInfo.FileName = vsWherePath;
+            StartInfo.Arguments = "-latest -requires Microsoft.Component.MSBuild -find MSBuild\\**\\Bin\\MSBuild.exe";
+
+            var vsWhereOutput = "";
+            var process = new Process { StartInfo = StartInfo };
+            process.OutputDataReceived += (sender, args) => vsWhereOutput += args.Data;
+            process.Start();
+            process.BeginOutputReadLine();
+            process.BeginErrorReadLine();
+            process.WaitForExit();
+
+            logger?.LogDebug("VSWhere returned with: {0}", vsWhereOutput);
+
+            if (!File.Exists(vsWhereOutput))
+            {
+                throw new FileNotFoundException($"Failed to locate 'msbuild'. VSWhere returned {vsWhereOutput}");
+            }
+
+            return vsWhereOutput;
+        }
+
+        internal static string LocateVSWhere(IDiagnosticLogger? logger)
+        {
+            var projectPath = Path.GetDirectoryName(Application.dataPath);
+            var directories = Directory.GetDirectories(Path.Combine(projectPath, "Library", "PackageCache"), "com.unity.ide.visualstudio@*");
+            if (directories is null || directories.Length < 1)
+            {
+                throw new Exception("Failed lo locate the 'com.unity.ide.visualstudio' package.");
+            }
+
+            // TODO: Not sure if there can be more than one version of a package in the Library/PackageCache and if it even matters
+            var vsPackagePath = directories[0];
+            logger?.LogDebug("Located 'com.unity.ide.visualstudio' package at {0}", vsPackagePath);
+
+            var vsWherePath = Path.Combine(vsPackagePath, "Editor", "VSWhere", "vswhere.exe");
+            if (!File.Exists(vsWherePath))
+            {
+                throw new FileNotFoundException($"Failed to find 'vswhere.exe' at '{vsWherePath}'");
+            }
+
+            return vsWherePath;
+        }
+
+        public static void Build(SentryUnityOptions options, string executablePath)
+        {
+            var vsWherePath = LocateVSWhere(options.DiagnosticLogger);
+            var msBuildPath = LocateMSBuild(vsWherePath, options.DiagnosticLogger);
+
+            var playerSource = LocateWindowsPlayerSource();
+            var playerTarget = FileUtil.GetUniqueTempPathInProject();
+
+            CreateWindowsPlayerProject(playerSource, playerTarget, options.DiagnosticLogger);
+
+            AddNativeOptions(options);
+            AddSentryToMain(options);
+
+            StartInfo.FileName = msBuildPath;
```

Comment:
We're not checking if the value was set before and skipping the vswhere call or any other optimization so modifying this static property here isn't giving us much and has side effects beyond this method invocation. Lets avoid it